x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/index.tsx

```diff
@@ -124,140 +79,97 @@ export function ServiceOverviewTransactionsTable(props: Props) {
           start,
           end,
           uiFilters: JSON.stringify(uiFilters),
-          size: PAGE_SIZE,
-          numBuckets: 20,
-          pageIndex: tableOptions.pageIndex,
-          sortField: tableOptions.sort.field,
-          sortDirection: tableOptions.sort.direction,
           transactionType,
           latencyAggregationType: latencyAggregationType as LatencyAggregationType,
         },
       },
-    }).then((response) => {
-      return {
-        items: response.transactionGroups,
-        totalItemCount: response.totalTransactionGroups,
-        tableOptions: {
-          pageIndex: tableOptions.pageIndex,
-          sort: {
-            field: tableOptions.sort.field,
-            direction: tableOptions.sort.direction,
-          },
-        },
-      };
     });
   }, [
     serviceName,
     start,
     end,
     uiFilters,
-    tableOptions.pageIndex,
-    tableOptions.sort.field,
-    tableOptions.sort.direction,
     transactionType,
     latencyAggregationType,
   ]);
 
+  const { transactionGroups, requestId } = data;
+  const sortedSlicedItems = orderBy(
+    transactionGroups,
+    sort.field,
+    sort.direction
+  ).slice(pageIndex * PAGE_SIZE, (pageIndex + 1) * PAGE_SIZE);
+
+  const transactionNames = useMemo(
+    () => sortedSlicedItems.map(({ name }) => name).join(),
+    [sortedSlicedItems]
```

Comment:
I think the other comment disappeared: what happens if a transaction name contains a comma?

Reply:
The transaction with a comma won't have the aggregation results shown in the table. We could join the names with something else, but we'd have to choose something very specific to avoid a collision. Do you have something in mind @sqren

Reply:
I don't think we need to memoize transaction names, pretty sure there's a better way to invalidate the second request

Reply:
@dgieselaar that is indeed no longer needed. Removing it.